File: backend/app/schemas/quality.py
"""Schemas for the data-quality report."""

from pydantic import BaseModel


class QualityIssue(BaseModel):
    id: str
    issue_type: str
    severity: str  # "error" | "warning"
    member_ids: list[str]
    description: str
    dismissed: bool = False


class QualityReport(BaseModel):
    tree_id: str
    total_members: int
    issues: list[QualityIssue]

File: frontend/src/types/quality.ts
export interface QualityIssue {
  id: string;
  issue_type: string;
  severity: "error" | "warning";
  member_ids: string[];
  description: string;
  dismissed: boolean;
}

export interface QualityReport {
  workspace_id: string;
  total_members: number;
  issues: QualityIssue[];
}
